plugins/inventory/opennebula.py

```diff
@@ -199,6 +199,8 @@ def _retrieve_servers(self, label_filter=None):
                     continue
 
             server['name'] = vm.NAME
+            server['id'] = vm.ID
+            server['host'] = vm.HISTORY_RECORDS.HISTORY[-1].HOSTNAME
```

Comment:
The unit tests fail because they don't know about `HISTORY` in `vm.HISTORY_RECORDS`.

First of all, in the unit test data, `HISTORY_RECORDS` is a dictionary, not an object, so `.HISTORY` will never work anyway. `vm.HISTORY_RECORDS['HISTORY']` would. Second, in the unit tests, the dictionary is an empty one, so even that would not work.

I don't know how the real data looks like, but I guess you will have to be more careful when extracting that information.

(Is it always guaranteed to be there, for example?)

Reply:
Hello guys, regarding HISTORY_RECORDS, there should always be some, but from my personal experience, after doing [database cleanup](https://docs.opennebula.io/6.8/installation_and_configuration/opennebula_services/database.html#onedb-purge-history) they can be empty, althrought, in documentation is written, that last two will be not deleted, bug?

Reply:
I also did some tests, and vminfopoolextended method is returning for each VM only last History record, for example:

```
<HISTORY_RECORDSTypeSub>
    <HISTORY>
        <OID>42</OID>
        <SEQ>834</SEQ>
        <HOSTNAME>dc6-node11.feldcloud.net</HOSTNAME>
        <HID>10</HID>
        <CID>0</CID>
        <DS_ID>100</DS_ID>
        <VM_MAD>kvm</VM_MAD>
        <TM_MAD>3par</TM_MAD>
        <ACTION>0</ACTION>
    </HISTORY>
</HISTORY_RECORDSTypeSub>
```

Reply:
Thank you very much for doing the tests @feldsam . Would you be able to help resolve the CI issue, please? It is beyond my level of python experience.

Reply:
@feldsam should I revert to the first commit where it was getting the host pool and mapping to the guests? Would mean there would always be a record available...

Reply:
I think, it can be good only as fallback, if there are no history records. Otherwise it is unnecessary more API requests. Also, it doesnt solve tests